launch/launch/actions/execute_process.py

```diff
@@ -316,46 +317,58 @@ def __on_process_stdin(
     def __on_process_stdout(
         self, event: ProcessIO
     ) -> Optional[SomeActionsType]:
-        output = (event.text.decode(errors='replace') + 'a').splitlines()
-        new_buffer = ''
-        # Check if the event text ended in a new line or not.
-        # If ended in a new line, print everything. If not, save the ending in the buffer.
-        if output[-1] != 'a':
-            new_buffer = output[-1][:-1]
-        output[0] = self.__stdout_buffer + output[0]
-        output = output[:-1]
-        self.__stdout_buffer = new_buffer
-        for line in output:
-            self.__stdout_logger.info(
-                self.__output_format.format(line=line, this=self)
-            )
+        with self.__stdout_buffer as buf:
```

Comment:
@ivanpauno what that does the context manager do here?

Reply:
I suppose it calls StringIO.close(). But I don't think it's really needed.